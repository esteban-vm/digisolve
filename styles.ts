import { css } from '@emotion/react'

export const globalStyles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --color-primary: #ed07b7;
    --color-dark-primary: #bd0592;
    --color-secondary: #6920b0;
    --color-tertiary: #acdaff;

    --color-white: #fff;
    --color-grey: #555;
    --color-black: #000;

    --color-light: #f4f4f4;
    --color-lighter: #fcfcfc;

    --color-overlay1: #0000004d;
    --color-overlay2: #000000b3;
    --color-overlay3: #00000080;
    --color-overlay4: #000000cc;

    --color-shadow: #efefefef;
  }

  html,
  body {
    background-color: var(--color-white);
    color: var(--color-black);
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }

  .clear_fix {
    zoom: 1;

    ::after {
      content: '';
      display: table;
      clear: both;
      visibility: hidden;
    }
  }
`

export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
