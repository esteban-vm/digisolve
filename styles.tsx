import { Global, css } from '@emotion/react'

const styles = css`
  @import url('/css/fluid.css');
  @import url('/css/normalize.min.css');

  @font-face {
    font-family: 'Work Sans';
    src: local('Work Sans'), url('/fonts/WorkSans-VariableFont_wght.ttf');
  }

  :root {
    --primary-color: #ed07b7;
    --primary-color-dark: #bd0592;
    --secondary-color: #6920b0;
    --light-color: #acdaff;
    --dark-color: #555;
  }

  html,
  body {
    background-color: #fff;
    color: #000;
    text-rendering: optimizeLegibility;
    font-size: 16px;
    font-family: 'Work Sans', Arial, sans-serif;
    font-weight: 300;
  }

  .row {
    max-width: 1140px;
    margin: 0 auto;
  }
`

export const Globals = () => <Global styles={styles} />
export * from '@emotion/react'
export { default as styled } from '@emotion/styled'
