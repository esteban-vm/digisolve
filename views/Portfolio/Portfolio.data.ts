import type { WorkProps } from './Work'

const width = 394
const height = 470

const works: WorkProps[] = [
  {
    id: '0a6dd16e-0229-4898-a3b4-b6b8a2dd5e5a',
    text: 'Online advertising',
    image: { src: '/img/online-advertising.png', width, height },
  },
  {
    id: '35501350-65b2-47bc-95a8-97bc27d2e186',
    text: 'Social media campaigns',
    image: { src: '/img/social-media-campaigns.png', width, height },
  },
  {
    id: '26c86632-8302-4ba4-b316-0a810712efef',
    text: 'UX design',
    image: { src: '/img/ux-design.png', width, height },
  },
  {
    id: 'e333c170-6d2a-43e9-9720-0a1d50d9c3ba',
    text: 'Branding & digital marketing',
    image: { src: '/img/branding-digital-marketing.png', width, height },
  },
]

export default works
