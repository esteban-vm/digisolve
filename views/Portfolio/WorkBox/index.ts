export { default, type WorkProps } from './WorkBox'
