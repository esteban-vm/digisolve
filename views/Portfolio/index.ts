export { default as Portfolio, works } from './Portfolio'
