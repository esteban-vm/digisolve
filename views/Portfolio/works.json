[
  {
    "title": "Online advertising",
    "image": {
      "src": "/img/online-advertising.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "Social media campaigns",
    "image": {
      "src": "/img/social-media-campaigns.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "UX design",
    "image": {
      "src": "/img/ux-design.png",
      "height": 470,
      "width": 394
    }
  },
  {
    "title": "Branding & digital marketing",
    "image": {
      "src": "/img/branding-digital-marketing.png",
      "height": 470,
      "width": 394
    }
  }
]
