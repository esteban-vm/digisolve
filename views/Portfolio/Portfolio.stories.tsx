import type { Meta } from '@storybook/react'
import Portfolio from './Portfolio'

export default {
  component: Portfolio,
  title: 'Views/Portfolio',
} as Meta<typeof Portfolio>

export const PortfolioView = () => <Portfolio />
