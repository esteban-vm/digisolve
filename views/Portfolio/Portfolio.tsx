import type { Component } from '@/types'
import { useId } from 'react'
import { Grid } from '@/components'
import { styled, mediaQuery } from '@/styles'
import works from './Portfolio.data'
import Work from './Work'

const PortfolioComponent: Component = (props) => {
  const nameID = useId()

  return (
    <section id='portfolio' aria-labelledby={nameID} {...props}>
      <Grid.Row>
        <h2 id={nameID} className='section__heading'>
          Our Work
        </h2>
        <h3 className='section__subheading'>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.
        </h3>
      </Grid.Row>
      <ul className='clear_fix'>
        {works.map((work) => (
          <li key={work.id}>
            <Work {...work} />
          </li>
        ))}
      </ul>
    </section>
  )
}

const Portfolio = styled(PortfolioComponent)`
  text-align: center;
  background: var(--color-light);

  ul {
    margin-top: 10px;
    width: 100%;

    li {
      display: block;
      width: 25%;
      float: left;

      ${mediaQuery('xs')} {
        width: 50%;
      }
    }
  }
`

export default Portfolio
