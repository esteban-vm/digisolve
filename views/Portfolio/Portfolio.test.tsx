import { render, cleanup, screen, create, axe } from '@/tests'
import { mediaQuery } from '@/styles'
import works from './Portfolio.data'
import Portfolio from './Portfolio'

describe('ðŸ§ª PORTFOLIO:', () => {
  describe('display tests:', () => {
    let container: HTMLElement

    beforeEach(() => {
      void ({ container } = render(<Portfolio />))
    })

    afterEach(cleanup)

    it('should be accessible', async () => {
      const results = await axe(container)
      expect(results).toHaveNoViolations()
    })

    describe('should be displayed:', () => {
      it('the title', () => {
        const title = screen.getByRole('heading', { name: /^our work$/i, level: 2 })
        expect(title).toBeInTheDocument()
        expect(title).toBeVisible()
      })

      it('the subtitle', () => {
        const subtitle = screen.getByRole('heading', { level: 3 })
        expect(subtitle).toBeInTheDocument()
        expect(subtitle).toBeVisible()
      })

      it('the works', () => {
        const workElements = screen.getAllByRole('figure')
        expect(workElements).toHaveLength(works.length)

        for (const work of workElements) {
          expect(work).toBeInTheDocument()
          expect(work).toBeVisible()
        }
      })
    })
  })

  describe('style tests:', () => {
    let tree: ReturnType<typeof create>

    beforeEach(() => {
      tree = create(<Portfolio />)
    })

    it('should render with correct styles', () => {
      expect(tree).toMatchSnapshot()
    })

    it('should have correct styles on extra small screen devices', () => {
      expect(tree).toHaveStyleRule('width', '50%', { target: 'ul li', media: mediaQuery('xs') })
    })
  })
})
