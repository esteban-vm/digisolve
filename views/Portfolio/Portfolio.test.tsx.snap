// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ðŸ§ª PORTFOLIO: style tests: should render with correct styles 1`] = `
.emotion-0 {
  text-align: center;
  background: var(--color-light);
}

.emotion-0 ul {
  margin-top: 10px;
  width: 100%;
}

.emotion-0 ul li {
  display: block;
  width: 25%;
  float: left;
}

@media only screen and (max-width: 480px) {
  .emotion-0 ul li {
    width: 50%;
  }
}

.emotion-1 {
  background-color: var(--color-black);
  overflow: hidden;
}

.emotion-1 img {
  width: 100%;
  height: auto;
  opacity: 0.7;
  -webkit-transform: scale(1.15);
  -moz-transform: scale(1.15);
  -ms-transform: scale(1.15);
  transform: scale(1.15);
  -webkit-transition: -webkit-transform 500ms,opacity 500ms;
  transition: transform 500ms,opacity 500ms;
}

.emotion-1 img:hover {
  -webkit-transform: scale(1);
  -moz-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
  opacity: 1;
}

.emotion-1 p {
  color: var(--color-white);
  text-transform: capitalize;
}

<section
  aria-labelledby=":r0:"
  className="emotion-0"
  id="portfolio"
>
  <div
    className="row "
    role="presentation"
  >
    <h2
      className="section__heading"
      id=":r0:"
    >
      Our Work
    </h2>
    <h3
      className="section__subheading"
    >
      Lorem ipsum dolor sit amet consectetur adipisicing elit. Omnis, ducimus.
    </h3>
  </div>
  <ul
    className="clear_fix"
  >
    <li>
      <figure
        aria-labelledby="0a6dd16e-0229-4898-a3b4-b6b8a2dd5e5a"
        className="emotion-1"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height={470}
          id="0a6dd16e-0229-4898-a3b4-b6b8a2dd5e5a"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fonline-advertising.png&w=828&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fonline-advertising.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fonline-advertising.png&w=828&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={394}
        />
        <p>
          Online advertising
        </p>
      </figure>
    </li>
    <li>
      <figure
        aria-labelledby="35501350-65b2-47bc-95a8-97bc27d2e186"
        className="emotion-1"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height={470}
          id="35501350-65b2-47bc-95a8-97bc27d2e186"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fsocial-media-campaigns.png&w=828&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fsocial-media-campaigns.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fsocial-media-campaigns.png&w=828&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={394}
        />
        <p>
          Social media campaigns
        </p>
      </figure>
    </li>
    <li>
      <figure
        aria-labelledby="26c86632-8302-4ba4-b316-0a810712efef"
        className="emotion-1"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height={470}
          id="26c86632-8302-4ba4-b316-0a810712efef"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fux-design.png&w=828&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fux-design.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fux-design.png&w=828&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={394}
        />
        <p>
          UX design
        </p>
      </figure>
    </li>
    <li>
      <figure
        aria-labelledby="e333c170-6d2a-43e9-9720-0a1d50d9c3ba"
        className="emotion-1"
      >
        <img
          alt=""
          data-nimg="1"
          decoding="async"
          height={470}
          id="e333c170-6d2a-43e9-9720-0a1d50d9c3ba"
          loading="lazy"
          onError={[Function]}
          onLoad={[Function]}
          src="/_next/image?url=%2Fimg%2Fbranding-digital-marketing.png&w=828&q=75"
          srcSet="/_next/image?url=%2Fimg%2Fbranding-digital-marketing.png&w=640&q=75 1x, /_next/image?url=%2Fimg%2Fbranding-digital-marketing.png&w=828&q=75 2x"
          style={
            {
              "color": "transparent",
            }
          }
          width={394}
        />
        <p>
          Branding & digital marketing
        </p>
      </figure>
    </li>
  </ul>
</section>
`;
